src/app/services/user-info-firebase.service.ts

```diff
@@ -4,29 +4,44 @@ import {
   Firestore, getFirestore,
   getDoc, setDoc, doc, collection,
   DocumentSnapshot } from '@angular/fire/firestore';
-import { Observable, defer, from, map, switchMap, of, iif } from 'rxjs';
+import { Observable, defer, from, map, switchMap, of, iif, take } from 'rxjs';
 import { UserData, UserModel } from '../data-models/user-model';
 
 @Injectable({
   providedIn: 'root'
 })
-export class UserInfoFirebaseService extends UserInfoService{
+export class UserInfoFirebaseService extends UserInfoService {
 
   constructor(private firestore: Firestore = getFirestore()) {
     super();
   }
 
   override loadUser(id: string): Observable<UserData> {
-    return defer(() => from(getDoc(doc(this.firestore, 'users', id)) as Promise<DocumentSnapshot>))
-      .pipe(switchMap((docSnap: DocumentSnapshot) =>
-        iif(() => !docSnap.exists(),
-          this.saveUser(new UserData(id, [], [])).pipe(map((dbUser) => new UserData(dbUser, [], []))),
-          of(docSnap.data() as UserModel).pipe(map((dbUser: UserModel) => UserData.copyUser(dbUser))
-        ))
-    ));
+    if (id === sessionStorage.getItem("userId")) {
+      const JSONuser = (JSON.parse(sessionStorage.getItem(id)!));
+      return of(new UserData(JSONuser.uid,JSONuser._recentSets,JSONuser._ownedSets));
+    } else {
+      sessionStorage.clear();
+      let userVar:UserData;
+      let userObservable =  defer(() =>
+        from(getDoc(doc(this.firestore, 'users', id)) as Promise<DocumentSnapshot>))
+        .pipe(switchMap((docSnap: DocumentSnapshot) =>
+          iif(() => !docSnap.exists(),
+            this.saveUser(new UserData(id, [], [])).pipe(map((dbUser) => new UserData(dbUser, [], []))),
+            of(docSnap.data() as UserModel).pipe(map((dbUser: UserModel) => UserData.copyUser(dbUser))
+          ))
```

Comment:
This doesn't appear to be saving the owned and viewed study sets of the user. 

Console log showing that the correct user data was fetched from the firestore, but empty arrays being created
<img width="1462" alt="image" src="https://github.com/PonderTeam/ponder-app/assets/4361319/e79188fb-fc4c-4847-886c-8c0169d49594">

On user cache hit after page reload, we get an empty page
<img width="1458" alt="image" src="https://github.com/PonderTeam/ponder-app/assets/4361319/2d5a385e-5798-4065-8d8b-81e5dced4bcc">